### PR TITLE
testing: cover chart widget behavior

### PR DESCRIPTION
**Pull request by RtlZeroMemory**
## Summary
- rewrite the weak text-mode chart widget tests around visible output for `sparkline`, `barChart`, and `miniChart`
- cover requested sparkline width, horizontal and vertical bar-chart text output, and `miniChart` bars vs pills variants with visible percentages
- keep high-resolution graphics transport, canvas/image behavior, and canvas-based chart surfaces out of scope because their strongest stable contracts are already lower-level

## Families and systems covered
- text-mode chart widgets: `sparkline`, `barChart`, and `miniChart`

## Tests added, rewritten, removed
- rewrote the text-mode chart portions of `packages/core/src/widgets/__tests__/dataVizWidgets.test.ts` from prop-shape checks to visible-output behavior assertions
- retained `lineChart`, `scatter`, and `heatmap` coverage in the same file because their strongest stable contracts remain lower-level graphics/protocol behavior

## Implementation bugs fixed
- none

## Validation
- `./node_modules/.bin/tsc -b packages/core/tsconfig.json --pretty false`
- `node --test packages/core/dist/widgets/__tests__/dataVizWidgets.test.js packages/core/dist/widgets/__tests__/basicWidgets.render.test.js packages/core/dist/widgets/__tests__/graphicsWidgets.test.js packages/core/dist/widgets/__tests__/graphics.golden.test.js`

## Explicit gaps left out
- `sparkline` and `barChart` high-resolution transport remain covered by retained graphics suites rather than this widget-render slice
- `canvas`, `image`, `lineChart`, `scatter`, and `heatmap` remain separate lower-level graphics/protocol surfaces
- this PR does not claim broader chart/media family completeness

## Summary by CodeRabbit

* **Tests**
  * Enhanced test coverage for data-visualization widgets (sparkline, barChart, miniChart) by switching to rendering-based assertions. Tests now validate user-facing textual output, glyphs, labels, percentages and trend/Bar visuals across edge cases and orientations to better ensure correct visual presentation.

### CODE REVIEW

File: packages/core/src/widgets/__tests__/dataVizWidgets.test.ts [hunk 1]
```diff
@@ -1,75 +1,90 @@
 import { assert, describe, test } from "@rezi-ui/testkit";
+import { createTestRenderer } from "../../testing/index.js";
```
[the remark this answers is not in the document]
Comment (RtlZeroMemory) [reply packages/core/src/widgets/__tests__/dataVizWidgets.test.ts:2]: These widget source tests already use the internal testing harness via relative imports across the suite, so switching only this file to `@rezi-ui/core/testing` would be inconsistent and does not strengthen a public behavior contract. Keeping the existing import path here.
